flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -731,9 +731,16 @@ public TableResultInternal executePlan(InternalPlan plan) {
     }
 
     private CompiledPlan compilePlanAndWrite(
-            String filePath, boolean ifNotExists, Operation operation) {
-        File file = Paths.get(filePath).toFile();
-        if (file.exists()) {
+            Path filePath, boolean ifNotExists, Operation operation) {
+        FileSystem fs;
+        boolean exists;
+        try {
+            fs = filePath.getFileSystem();
+            exists = fs.exists(filePath);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
I think we'd better throw `TableException` instead of `RuntimeException`

Reply:
Okey. I have modified this in the latest commit.